spec/calendar.html

```diff
@@ -313,189 +387,354 @@ <h1>get Temporal.Calendar.prototype.id</h1>
+      </p>
+      <p>
+        The `daysInYear` method takes one argument _dateOrDateTime_.
         The following steps are taken:
       </p>
       <emu-alg>
-        1. Throw an *Error* exception.
+        1. Let _calendar_ be the *this* value.
+        1. Perform ? RequireInternalSlot(_calendar_, [[InitializedTemporalCalendar]]).
+        1. If _calendar_.[[Identifier]] is not *"iso8601"*, throw an *Error* exception.
+        1. If _dateOrDateTime_ does not have an [[ISOYear]] internal slot, then
+          1. Set _dateOrDateTime_ to ? ToTemporalDate(_dateOrDateTime_).
+        1. Return 𝔽(! DaysInYear(_dateOrDateTime_.[[ISOYear]])).
       </emu-alg>
     </emu-clause>
 
     <emu-clause id="sec-temporal.calendar.prototype.monthsinyear">
-      <h1>Temporal.Calendar.prototype.monthsInYear ( _date_ )</h1>
+      <h1>Temporal.Calendar.prototype.monthsInYear ( _dateOrDateTime_ )</h1>
+      <p>
+        An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement the `monthsInYear` method as specified in the ECMA-402 specification.
+        If an ECMAScript implementation does not include the ECMA-402 API the following specification of the `monthsInYear` method is used.
+      </p>
       <p>
-        The `monthsInYear` method takes one argument _date_.
+        The `monthsInYear` method takes one argument _dateOrDateTime_.
         The following steps are taken:
       </p>
       <emu-alg>
-        1. Throw an *Error* exception.
+        1. Let _calendar_ be the *this* value.
+        1. Perform ? RequireInternalSlot(_calendar_, [[InitializedTemporalCalendar]]).
+        1. If _calendar_.[[Identifier]] is not *"iso8601"*, throw an *Error* exception.
+        1. Perform ? ToTemporalDateTime(_dateOrDateTime_).
+        1. Return *12*<sub>𝔽</sub>.
       </emu-alg>
     </emu-clause>
 
     <emu-clause id="sec-temporal.calendar.prototype.inleapyear">
-      <h1>Temporal.Calendar.prototype.inLeapYear ( _date_ )</h1>
+      <h1>Temporal.Calendar.prototype.inLeapYear ( _dateOrDateTime_ )</h1>
       <p>
-        The `inLeapYear` method takes one argument _date_.
+        An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement the `inLeapYear` method as specified in the ECMA-402 specification.
+        If an ECMAScript implementation does not include the ECMA-402 API the following specification of the `inLeapYear` method is used.
+      </p>
+      <p>
+        The `inLeapYear` method takes one argument _dateOrDateTime_.
         The following steps are taken:
       </p>
       <emu-alg>
-        1. Throw an *Error* exception.
+        1. Let _calendar_ be the *this* value.
+        1. Perform ? RequireInternalSlot(_calendar_, [[InitializedTemporalCalendar]]).
+        1. If _calendar_.[[Identifier]] is not *"iso8601"*, throw an *Error* exception.
```

Comment:
I guess this line and other similar lines should be an assertion, as it's not possible to construct a calendar which isn't iso8601.